code/__DEFINES/dcs/signals.dm

```diff
@@ -586,3 +586,8 @@
 // /datum/action signals
 #define COMSIG_ACTION_TRIGGER "action_trigger"                        //from base of datum/action/proc/Trigger(): (datum/action)
 	#define COMPONENT_ACTION_BLOCK_TRIGGER (1<<0)
+
+//Signals for automatic fire at component
+#define COMSIG_START_SHOOTING_AT 	"start_shooting_at"
+#define COMSIG_STOP_SHOOTING_AT		"stop_shooting_at"
+#define COMSIG_SHOT_FIRED			"shot_fired"
```

Comment:
Usually it is typical to explain which component this is applicable for
For instance:
COMSIG_START_SHOOTING_AT -> COMSIG_AUTOMATIC_SHOOT_START_AT